pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -377,6 +388,40 @@ func (p *provider) parseTopologySpreadConstraint(topologyConstraints []kubevirtt
 	return parsedTopologyConstraints, nil
 }
 
+func (p *provider) parseOSImage(primaryDisk kubevirttypes.PrimaryDisk, nameSpace string) (*OSImage, error) {
+	osImage, err := p.configVarResolver.GetConfigVarStringValue(primaryDisk.OsImage)
+	if err != nil {
+		return nil, fmt.Errorf(`failed to get value of "primaryDisk.osImage" field: %w`, err)
+	}
+
+	var parsedOSImage *OSImage
+	// If PrimaryDiskOSSource is specified
+	if primaryDisk.Source != nil {
+		if primaryDisk.Source.HTTP != nil {
+			preAllocatedDV, _, err := p.configVarResolver.GetConfigVarBoolValue(primaryDisk.Source.HTTP.PreAllocateDataVolume)
+			if err != nil {
+				return nil, fmt.Errorf(`failed to get value of "preAllocateDataVolume" field: %w`, err)
+			}
+			parsedOSImage = &OSImage{HTTP: &HTTPSource{URL: osImage, PreAllocatedDataVolume: preAllocatedDV}}
+		} else if primaryDisk.Source.PVCClone != nil {
+			parsedOSImage = &OSImage{PVCClone: &PVCSource{DataVolumeName: osImage, NameSpace: nameSpace}}
+		}
+	}
+	// If PrimaryDiskOSSource is not specified
```

Comment:
We have to extend that comment and say that we are dealing here with older API and it's a matter of backward compatibility